File: lab1/e.py
import queue

borisCards= queue.Queue()
nurikCards= queue.Queue()

for i in range(0, 10):
    card = int(input())
    if i< 5:
        borisCards.put(card)
    else:
        nurikCards.put(card)

cnt = 0

while not borisCards.empty() and not nurikCards.empty():
    cnt += 1
    if cnt == 1000000:
       print("blin nichya")
       break

    if borisCards.front()== 0 and nurikCards.front()== 9:
        borisCards.put(borisCards.front())
        borisCards.put(nurikCards.front())
        borisCards.pop()
        nurikCards.pop()

    elif borisCards.front()== 9 and nurikCards.front()== 0:
        nurikCards.put(borisCards.front())
        nurikCards.put(nurikCards.front())
        borisCards.pop()
        nurikCards.pop()
    elif borisCards.front()> nurikCards.front():
        borisCards.put(borisCards.front())
        borisCards.put(nurikCards.front())
        borisCards.pop()
        nurikCards.pop()
    elif borisCards.front()< nurikCards.front():
        nurikCards.put(borisCards.front())
        nurikCards.put(nurikCards.front())
        borisCards.pop()
        nurikCards.pop()

    if nurikCards.empty():
        print("Boris ")
    else:
        print("Nursik ")

File: lab1/e.cpp
#include <iostream>
#include <queue>
using namespace std; 
int main() {
    // Your code goes here
    queue<int> boris; 
    queue<int> nursik; 

    for(int i=0; i <10; i++){ 
        int card; cin>> card; 
        if(i<5){ 
            boris.push(card);
        }
        else{ 
            nursik.push(card); 
        }
    }

    int cnt=0; 
    while(!boris.empty()&& !nursik.empty()){ 
        cnt++; 
        if(cnt==100000){ 
            cout<< "blin nichiya"<<endl; 
            break;
        }

        if(boris.front() == 0 && nursik.front() == 9){ 
            boris.push(boris.front()); 
            boris.push(nursik.front()); 
            boris.pop(); 
            nursik.pop();
        }
        else if(boris.front() == 9 && nursik.front() == 0){ 
            nursik.push(boris.front()); 
            nursik.push(nursik.front()); 
            boris.pop(); 
            nursik.pop();
        }
        else if(boris.front() > nursik.front()){ 
            boris.push(boris.front()); 
            boris.push(nursik.front()); 
            boris.pop(); 
            nursik.pop();
        }
        else{ 
            nursik.push(boris.front()); 
            nursik.push(nursik.front()); 
            boris.pop(); 
            nursik.pop();
        }
    }
    if(nursik.empty()){ 
        cout<< "Boris "<< cnt; 
    }
    else{ 
        cout<< "Nursik "<< cnt;
    }
}